packages/amplify_api/lib/src/graphql/paginated_modelType_impl.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api/src/graphql/paginated_result_impl.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+class PaginatedModelTypeImpl<T extends Model> extends PaginatedModelType<T> {
+  final ModelType<T> modelType;
+
+  PaginatedModelTypeImpl(this.modelType) : super(modelType);
+
+  @override
+  PaginatedResultImpl<T> fromJson(Map<String, dynamic> jsonData) {
+    List<dynamic>? dataList = List<dynamic>.from(jsonData['items']);
+
+    List<T> decodedData = dataList.map((e) => modelType.fromJson(e)).toList();
+
+    return PaginatedResultImpl<T>(decodedData, jsonData['nextToken']);
+  }
```

Comment:
There are a few implicit casts throughout here which we need to be careful of. Here's how I would do it to avoid null unwraps or bad casts:

```dart
  @override
  PaginatedResultImpl<T> fromJson(Map<String, dynamic> jsonData) {
    final itemsJson = jsonData['items'] as List?;

    if (itemsJson == null || itemsJson.isEmpty) {
      return const PaginatedResultImpl<T>([], null);
    }

    final items = itemsJson
        .cast<Map>()
        .map(
          (e) => modelType.fromJson(e.cast()),
        )
        .toList();

    return PaginatedResultImpl<T>(items, jsonData['nextToken'] as String?);
  }
```

Reply:
Slick! Thanks for the tip.